src/auth/ducks/thunks.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  LoginRequest,
+  SignupRequest,
+  TokenResponse,
+  UserAuthenticationThunkAction,
+} from './types';
+import { authenticateUser, logoutUser } from './actions';
+
+export const login = (
+  loginRequest: LoginRequest,
+): UserAuthenticationThunkAction<void> => {
+  return (dispatch, getState, { authClient, tokenService }): Promise<void> => {
+    return authClient
+      .login(loginRequest)
+      .then((response: TokenResponse) => {
+        tokenService.setAccessToken(response.accessToken);
```

Comment:
It seems weird that we have this "tokenService" file that is storing our tokens if we're now also storing the token information in state. Would it make sense to consolidate where we keep that data?

Reply:
tokenService is separated purely because it relies upon local storage. At first, I thought keeping the tokens in Redux may be insecure or bad practice, but either way they'll be easily accessible in the browser once you've logged in, which isn't necessarily a problem. I think it'd be a lot easier to have in the Redux state, we just have to account for storing the refresh token in localStorage and accessing it upon application load.